### DISCUSSION TITLE
Two RustDesk Clients ID@rdrelay?=key not working

### DISCUSSION THREAD

**Discussion by devNull2and1** [rustdesk/rustdesk#6517] [Q&A]
Hello we switched completly from teamviewer to rust desk pro now and got trouble on first day. Customer has also rustdesk installed with own relay, we are not able to connect to clients there. our rd.exe with compiled rdrelay connecting to his server and not to ours. Connecting from our side with ID@rdrelay?=key over his relayserver does not work too.

There must be a solution. Read somewhere passing -qs- to the filename maybe uses quick support, but no! How the **** can we solve this? :)

**Comment by rustdesk** [comment]
`ID@rdrelay?=key` this feature only availabe in nightly build.

**Comment by devNull2and1** [comment]
okay, we have generated our.exe with flutter nightly build...

We have this Options:
- Tell Customer to build their.exe with nightly build
- Telling our coworkers, dont type the ID for connection, you must type 1267423786@1.2.3.4?key=23ft67gnxc4f2tgn8y24frf678 this a bit complicated on the phone
- Tell our customers to uninstall rustdesk and connect

There is no other way? No possibility to start two RustDesk.exe one connecting to their relay and ours to our relay?

**Reply by rustdesk** [reply]
Sorry, we have no other way yet. At least for now.

**Comment by devNull2and1** [comment]
No Problem, for now. Im writing to inform you. Maybe we think about a solution for everyone that fits all needs. We boughht pro. Just take our money :)

Will not spent money for teamviewer anymore.

**Reply by rustdesk** [reply]
Of course, let's discuss with all users to get a common and good solution. `ID@rdrelay?=key` is just the first start and workaround, also suggested by users.

**Comment by devNull2and1** [comment]
> Of course, let's discuss with all users to get a common and good solution.

Problem is that this is a common problem when using rustdesk for customer support. I think this should go with quicksupport and branded settings like, show cursor on both sides, always scale auto and be able to open rustdesk with own connection to our relay while other local installation of rustdesk are used to connect their relay.
Somtimes you have to force desicions without wasting time and discuss with all.

**Reply by rustdesk** [reply]
Frankly, we thought less about customer support scenario for now, we thought more for RMM/MSP scenario in the past.

> open rustdesk while other local installation of rustdesk is used.

This is helpful, I do not know this if you do not tell me. That's why we need to discuss with users. I am not the real user, I do not know many things.

**Reply by rustdesk** [reply]
@21pages while you are working on `-qs-` version, please take into account of `open rustdesk while other local installation of rustdesk is used.`. If there is a local installation and `--server` process connecting to different self-hosting server, we need to open another server ipc (let's use --server-qs) and save config in different directory (not overwrite current user's config)

**Reply by devNull2and1** [reply]
I don't want to be cheeky, but I have a lot of ideas and as a company we are willing to pay more annually so that you can develop the features. We have the smallest subscription for the equivalent of €240 per year. We would have no problems paying €500 per year When features are improved, another big problem is port 21114-21119, there needs to be a way to get this to port 80 and 443 because many customers have ports blocked in their networks

**Reply by devNull2and1** [reply]
i dont talk about that this features must be available for all, maybe just the pro version. the ability to compile quicksupport with all possible configurations and co exist while ohter localrustdesk is running and this port 80 and 443 thing would be unique selling points against teamviewer, anydesk and other money greedy companies

**Reply by rustdesk** [reply]
How about udp? since we are still relying on one udp port.

**Reply by devNull2and1** [reply]
im mixing up many technical things now, im sorry. already tried to force hbr/hbs services to port 80/443, it was working on windows but not android. maybe i should split my issues. I just wanted to give a few examples...

**Reply by rustdesk** [reply]
Thanks, you are advanced user, giving many good ideas. We will make hbbr/hbbs using port 80/443 easier for normal users.

**Reply by rustdesk** [reply]
But if hbbs/hbbr use 80/443 port, the api port (for web concole) will be a problem. How do you handle this?

**Reply by devNull2and1** [reply]
Second IP Adress...We have dedicated servers and subnets on Hetzner like you. 

rdrelay.domain.com:80 -> 1.2.3.4 -> docker:21116
rdidserver.domain.com:443 -> 1.2.3.4 -> docker:21117
rdapi.domain.com:80 -> 1.2.3.5 -> docker:21114

just examples....again

i tested this but i did not work....was going to use wireshark, netstat to see connections and somehow android client has used rdrelay.domain.com:88 instead of 80. dont know why. in client config it was forced to rdrelay.domain.com:80 builded with flutter nightly

**Reply by rustdesk** [reply]
Got it

**Reply by rustdesk** [reply]
I will keep thinking about this to make using 443/80 easier.

**Reply by devNull2and1** [reply]
Maybe i did something wrong, i try to understand docker ports 21114-21119. Which one are the important ones?. As far as i understood. 2114 ist console/api.

When i map 80 to tcp/udp 21117 thats relay?
When i map 443 to tcp 21115 thats ID? 

for what is 21116,21118 and 21119?

**Reply by rustdesk** [reply]
21114 for console/api, 21116 udp/tcp for hbbs (ID server), 21117 for hbbr (relay), 21115 is a help port for NAT detection, 21118 for web socket of hbbs, 21119 for web socket of hbbr, if you do not use web client (not updated for long, but still work), 21118/21119 is useless. 21115 is not a must (maybe cause using relay always).

**Reply by rustdesk** [reply]
Here is document https://rustdesk.com/docs/en/self-host/rustdesk-server-oss/install/#step-2-run-hbbs-and-hbbr-on-your-server

![image](https://github.com/rustdesk/rustdesk/assets/71636191/4f081da3-5485-4140-98a1-b0b523cf20c9)

**Reply by devNull2and1** [reply]
When i have to use 2 IP Adresses i can map this too. But on client side it must be configured too.
I think that android version did not use the port i telled him.

Server:
```
rdrelay.domain.com:80 -> 1.2.3.4 -> docker:21117
rdidserver.domain.com:443 -> 1.2.3.4 -> docker:21116
rdapi.domain.com:80 -> 1.2.3.5 -> docker:21114
rdnat.domain.com:443 -> 1.2.3.5 -> docker:21115
```

Client:
```
relay address -> rdrelay.domain.com:80
id address -> rdidserver.domain.com:443
api address -> rdapi.domain.com:80
nat address -> rdnat.domain.com:443
```

> Here is document https://rustdesk.com/docs/en/self-host/rustdesk-server-oss/install/#step-2-run-hbbs-and-hbbr-on-your-server

Already found and read it, the linking of hbbr and hbbs as a ID/Relay and the configuration on the clients was not immediately clear to me

**Reply by rustdesk** [reply]
relay address  can be configured on web console, client do not need to know this.

**Reply by rustdesk** [reply]
I can not confirm if really worth config nat address on client side. We plan to remove this nat port to use a clever solution, sadly we have no resource to do this yet.

**Reply by devNull2and1** [reply]
I will try this again and just add a second relay to config for testing and come back to discussion then.

**Comment by devNull2and1** [comment]
@rustdesk 7 months later and many things changed. the new custom client solves our problem, now we can generate exe for quick support. even when our customer have their own rustdesk we are able to connect. we did not test it in detail but the time for it has come. also android custom client is there. thank you for this big gamechange. we will now spent time to configure everything we need.

but one final thing is open. this port behavior. i am not able to set the id server, it is missing on custom settings:

![image](https://github.com/rustdesk/rustdesk/assets/144108510/c4083907-daab-49a2-b145-10652e5cfec4)

at the moment it is like this:
myrelay.domain.com:443 -> 1.2.3.4 -> docker:21117
console.domain.com:443 -> 4.3.2.1 -> docker:21114

missing:
idserver.domain.com:443 -> 5.4.3.2 -> docker:21116

im right or did things change. id server no longer needed under this port?

**Reply by devNull2and1** [reply]
seems wrong:
myrelay.domain.com:443 -> 1.2.3.4 -> docker:21117 

works:
myrelay.domain.com:443 -> 1.2.3.4 -> docker:21116

but what is this port 442, somehow rd client try to connect but firewall blocks it. but connection works.
myrelay.domain.com:44444 -> 1.2.3.4 -> docker:21116
then client try to connect 44443

![image](https://github.com/rustdesk/rustdesk/assets/144108510/74dd5420-e552-423b-9920-d96c82484842)

**Reply by rustdesk** [reply]
`Host` on custom client of web console is the `ID server` on settings -> network page of rustdesk client.

**Reply by xlionjuan** [reply]
I'm guessing port `442` is the same thing of the original port `21115` (21116-1)?

**Reply by devNull2and1** [reply]
> `Host` on custom client of web console is the `ID server` on settings -> network page of rustdesk client.

Thx for answer, the last step and final missing part is to get rid of the ports 21114-21119. They are blocked in many networks.
I have no success till now. It would and should be possible when i can set relay, id and api domain+port over new custom settings like this:

```
relay.domain.com:443 -> 1.2.3.4 -> docker:21117
id.domain.com:443 -> 5.4.3.2 -> docker:21116
api.domain.com:443 -> 8.7.6.5 -> docker:21114
```
With this soultion i need 3 static and public ip adresses but for the moment it is okay for me. No problem.
In the future, maybe it is possible to handle everything over one port.

> I'm guessing port `442` is the same thing of the original port `21115` (21116-1)?

i don't understand the design and why this should be necessary?

EDIT: it takes relay from console relay settings, deletet all other relays, compiled it and it is branded in the exe. after that testet with wireshark on the client that connect and on the client that receive. he tries to connect but it dont work. there is always a try for 21115,21117 or 21116 which is blocked like in many enterprises.

**Reply by devNull2and1** [reply]
After 3 hours I've given up. You have such great software and it's really not far off until you can use it for remote support instead of TeamViewer or Anydesk. At this point in time it's still very frustrating for us. There is no way to work with port 443 and in many networks 21114-21119 is blocked. For us it's an absolute no-go to continue testing.

Overwriting the settings work but is not yet suitable for everyday use on the front with the dumbest user imaginable. See:
https://github.com/rustdesk/rustdesk/issues/8299

**Reply by xlionjuan** [reply]
`21115` is for NAT type test according to their doc
